service/matching/matching_engine_test.go

```diff
@@ -3015,6 +3434,22 @@ func (m *testTaskManager) DeleteTaskQueue(
 	return nil
 }
 
+// generateErrorRandomly states if a taskManager's operation should return an error or not
+func (m *testTaskManager) generateErrorRandomly() bool {
+	if m.dbError {
+		threshold := 10
+
+		// Seed the random number generator
+		r := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
don't re-seed the generator for each number, just get numbers from the global generator. seeding is handled automatically at startup in tests (and also non-tests now)

Reply:
curious by what you mean by the "global" generator here - for what it's worth, I changed the current implementation to now have a generator object for every test and don't re-seed it for each number. Let me know if this works.

Reply:
I commented elsewhere.. use `rand.Intn`. you don't need a custom source or generator